### PR TITLE
Update agent model

### PR DESCRIPTION
**Pull request by GGP1**
|Related issue|
|---|
| Closes https://github.com/wazuh/wazuh/issues/25582 |

## Description

Updates the indexer module Agent model according to the changes in the [index template](https://github.com/wazuh/wazuh-indexer/issues/270#issuecomment-2176507169).

## Tests

> [!Note]
> The new fields aren't set during the agent creation, that's why I indexed three agents to filter by those fields.

<details><summary>Agent creation</summary>

```console
gasti@gasti:~/work/wazuh$ curl -X POST -k -s -H  "Authorization: Bearer $TOKEN" https://localhost:55050/agents -H 'Content-Type: application/json' -d '{"id": "0191be8b-dbdd-7ce7-afd2-4d81993f73c5", "name": "test", "key": "598847fb25d20f5084a9dea56d3448e3"}'
{
  "data": {
    "id": "0191be8b-dbdd-7ce7-afd2-4d81993f73c5",
    "name": "test",
    "key": "\u00f7\u0094\u0097\u0013a\"\u00a7}\u00d1\u00c5\u00ec%\u0018\u00a79X\u00c8\u0018D\u007f\u008e\u00b3\u0093\u00ef\u00d3e\u0014\u00ed\u00a2\u0006\u00f4\u00ed\u00a7\u00f9f{$\u00b33LU\\2\u00fb\u0015\u00fdj\u0081"
  },
  "error": 0
}
```

</details>

<details><summary>Agents indexing</summary>

```console
root@wazuh-master:/# curl -X POST -H "Content-Type: application/json" -ku admin:SecretPassword1% http://wazuh-indexer:9200/agents/_create/0192009d-c049-767f-bd8e-ddff08ae03db?pretty=true -d '
{
  "id": "0192009d-c049-767f-bd8e-ddff08ae03db",
  "name": "test1",
  "is_connected": true,
  "host": {
    "ip": "192.168.0.1",
    "os": {
      "full": "Ubuntu 24.04.1 LTS (Noble Numbat)"
    }
  }
}
'
{
  "_index" : "agents",
  "_id" : "0192009d-c049-767f-bd8e-ddff08ae03db",
  "_version" : 1,
  "result" : "created",
  "_shards" : {
    "total" : 2,
    "successful" : 1,
    "failed" : 0
  },
  "_seq_no" : 3,
  "_primary_term" : 1
}
root@wazuh-master:/# curl -X POST -H "Content-Type: application/json" -ku admin:SecretPassword1% http://wazuh-indexer:9200/agents/_create/01912d07-b9b4-7528-bdad-15da902d651c?pretty=true -d '
{
  "id": "01912d07-b9b4-7528-bdad-15da902d651c",
  "name": "test2",
  "is_connected": false,
  "host": {
    "ip": "192.168.0.1",
    "os": {
      "full": "Ubuntu 23.10 (Mantic Minotaur)"
    }
  }
}
'
{
  "_index" : "agents",
  "_id" : "01912d07-b9b4-7528-bdad-15da902d651c",
  "_version" : 1,
  "result" : "created",
  "_shards" : {
    "total" : 2,
    "successful" : 1,
    "failed" : 0
  },
  "_seq_no" : 4,
  "_primary_term" : 1
}
root@wazuh-master:/# curl -X POST -H "Content-Type: application/json" -ku admin:SecretPassword1% http://wazuh-indexer:9200/agents/_create/0191fff9-a0d7-7917-b293-10298658f832?pretty=true -d '
{
  "id": "0191fff9-a0d7-7917-b293-10298658f832",
  "name": "test3",
  "is_connected": false,
  "host": {
    "ip": "192.168.0.1",
    "os": {
      "full": "Ubuntu 22.04.02 LTS (Jammy Jellyfish)"
    }
  }
}
'
{
  "_index" : "agents",
  "_id" : "0191fff9-a0d7-7917-b293-10298658f832",
  "_version" : 1,
  "result" : "created",
  "_shards" : {
    "total" : 2,
    "successful" : 1,
    "failed" : 0
  },
  "_seq_no" : 5,
  "_primary_term" : 1
}
```

</details>

<details><summary>Agent listing</summary>

All agents

```console
gasti@gasti:~/work/wazuh$ curl -k -s -H  "Authorization: Bearer $TOKEN" 'https://localhost:55050/agents?pretty=true'
{
   "data": {
      "affected_items": [
         {
            "id": "0191be8b-dbdd-7ce7-afd2-4d81993f73c5",
            "name": "test"
         },
         {
            "id": "0192009d-c049-767f-bd8e-ddff08ae03db",
            "name": "test1",
            "is_connected": true,
            "host": {
               "os": {
                  "full": "Ubuntu 24.04.1 LTS (Noble Numbat)"
               },
               "ip": "192.168.0.1"
            }
         },
         {
            "id": "01912d07-b9b4-7528-bdad-15da902d651c",
            "name": "test2",
            "is_connected": false,
            "host": {
               "os": {
                  "full": "Ubuntu 23.10 (Mantic Minotaur)"
               },
               "ip": "192.168.0.1"
            }
         },
         {
            "id": "0191fff9-a0d7-7917-b293-10298658f832",
            "name": "test3",
            "is_connected": false,
            "host": {
               "os": {
                  "full": "Ubuntu 22.04.02 LTS (Jammy Jellyfish)"
               },
               "ip": "192.168.0.1"
            }
         }
      ],
      "total_affected_items": 4,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected agents information was returned",
   "error": 0
}
```

Filter by is_connected

```console
gasti@gasti:~/work/wazuh$ curl -k -s -H  "Authorization: Bearer $TOKEN" 'https://localhost:55050/agents?pretty=true&is_connected=true'
{
   "data": {
      "affected_items": [
         {
            "id": "0192009d-c049-767f-bd8e-ddff08ae03db",
            "name": "test1",
            "is_connected": true,
            "host": {
               "os": {
                  "full": "Ubuntu 24.04.1 LTS (Noble Numbat)"
               },
               "ip": "192.168.0.1"
            }
         }
      ],
      "total_affected_items": 1,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected agents information was returned",
   "error": 0
}
```

Filter by host IP

```console
gasti@gasti:~/work/wazuh$ curl -k -s -H  "Authorization: Bearer $TOKEN" 'https://localhost:55050/agents?pretty=true&remote.ip=192.168.0.1'
{
   "data": {
      "affected_items": [
         {
            "id": "0191be8b-dbdd-7ce7-afd2-4d81993f73c5",
            "name": "test"
         },
         {
            "id": "0192009d-c049-767f-bd8e-ddff08ae03db",
            "name": "test1",
            "is_connected": true,
            "host": {
               "os": {
                  "full": "Ubuntu 24.04.1 LTS (Noble Numbat)"
               },
               "ip": "192.168.0.1"
            }
         },
         {
            "id": "01912d07-b9b4-7528-bdad-15da902d651c",
            "name": "test2",
            "is_connected": false,
            "host": {
               "os": {
                  "full": "Ubuntu 23.10 (Mantic Minotaur)"
               },
               "ip": "192.168.0.1"
            }
         },
         {
            "id": "0191fff9-a0d7-7917-b293-10298658f832",
            "name": "test3",
            "is_connected": false,
            "host": {
               "os": {
                  "full": "Ubuntu 22.04.02 LTS (Jammy Jellyfish)"
               },
               "ip": "192.168.0.1"
            }
         }
      ],
      "total_affected_items": 4,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected agents information was returned",
   "error": 0
}
gasti@gasti:~/work/wazuh$ curl -k -s -H  "Authorization: Bearer $TOKEN" 'https://localhost:55050/agents?pretty=true&remote.ip=192.168.0.2'
{
   "data": {
      "affected_items": [],
      "total_affected_items": 0,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "No agent information was returned",
   "error": 0
}
```

Filter by OS

```console
gasti@gasti:~/work/wazuh$ curl -k -H  "Authorization: Bearer $TOKEN" 'https://localhost:55050/agents?pretty=true&os.full=24.04.1'
{
   "data": {
      "affected_items": [
         {
            "id": "0192009d-c049-767f-bd8e-ddff08ae03db",
            "name": "test1",
            "is_connected": true,
            "host": {
               "os": {
                  "full": "Ubuntu 24.04.1 LTS (Noble Numbat)"
               },
               "ip": "192.168.0.1"
            }
         }
      ],
      "total_affected_items": 1,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected agents information was returned",
   "error": 0
}
gasti@gasti:~/work/wazuh$ curl -k -H  "Authorization: Bearer $TOKEN" 'https://localhost:55050/agents?pretty=true&os.full=23.10'
{
   "data": {
      "affected_items": [
         {
            "id": "01912d07-b9b4-7528-bdad-15da902d651c",
            "name": "test2",
            "is_connected": false,
            "host": {
               "os": {
                  "full": "Ubuntu 23.10 (Mantic Minotaur)"
               },
               "ip": "192.168.0.1"
            }
         }
      ],
      "total_affected_items": 1,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected agents information was returned",
   "error": 0
}
```

</details>

<details><summary>Agent deletion</summary>

By ID

```console
gasti@gasti:~/work/wazuh$ curl -X DELETE -k -H "Authorization: Bearer $TOKEN" 'https://localhost:55050/agents?agents_list=0191be8b-dbdd-7ce7-afd2-4d81993f73c5&pretty=true'
{
   "data": {
      "affected_items": [
         "0191be8b-dbdd-7ce7-afd2-4d81993f73c5"
      ],
      "total_affected_items": 1,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected agents were deleted",
   "error": 0
}
```

By is_connected

```console
gasti@gasti:~/work/wazuh$ curl -X DELETE -k -s -H  "Authorization: Bearer $TOKEN" 'https://localhost:55050/agents?agents_list=all&is_connected=true&pretty=true'
{
   "data": {
      "affected_items": [
         "0192009d-c049-767f-bd8e-ddff08ae03db"
      ],
      "total_affected_items": 1,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected agents were deleted",
   "error": 0
}
```

By OS

```console
gasti@gasti:~/work/wazuh$ curl -X DELETE -k -s -H  "Authorization: Bearer $TOKEN" 'https://localhost:55050/agents?agents_list=all&os.full=23.10&pretty=true'
{
   "data": {
      "affected_items": [
         "01912d07-b9b4-7528-bdad-15da902d651c"
      ],
      "total_affected_items": 1,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected agents were deleted",
   "error": 0
}
```

By IP

```console
gasti@gasti:~/work/wazuh$ curl -X DELETE -k -s -H  "Authorization: Bearer $TOKEN" 'https://localhost:55050/agents?agents_list=all&remote.ip=192.168.0.1&pretty=true'
{
   "data": {
      "affected_items": [
         "0191fff9-a0d7-7917-b293-10298658f832"
      ],
      "total_affected_items": 1,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected agents were deleted",
   "error": 0
}
```

</details>

### CODE REVIEW

File: framework/wazuh/core/indexer/events.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ async def create(self, events: List[StatefulEvent]) -> dict:
         # TODO(#24713): Implement server to indexer events batching
         response = {}
         for i, event in enumerate(events.events):
-            resp = await self._client.create(index=event.get_index_name(), body=asdict(event))
+            resp = await self._client.index(index=event.get_index_name(), body=asdict(event))
```
Comment (GGP1) [inline framework/wazuh/core/indexer/events.py:27]: This is a typo I noticed during the tests, I figured we could include it directly in this PR

File: api/api/controllers/agent_controller.py [hunk 2]
```diff
@@ -74,7 +73,9 @@ async def delete_agents(
             'type': type,
             'version': version,
             'last_login': older_than,
-            'persistent_connection_node': node_name,
+            'is_connected': is_connected,
+            'host.ip': request.query_params.get('remote.ip', None),
+            'host.os.full': request.query_params.get('os.full', None),
```
Comment (nico-stefani) [inline api/api/controllers/agent_controller.py:78]: Why are you getting the parameters from the request and not as arguments of the controller? I'd normalize this to don't mix.

File: api/api/controllers/agent_controller.py [hunk 3]
```diff
@@ -157,7 +158,9 @@ async def get_agents(
...
```
Comment (nico-stefani) [inline api/api/controllers/agent_controller.py:163]: Same here

File: api/api/controllers/agent_controller.py (hunk 2 above)
Comment (GGP1) [reply api/api/controllers/agent_controller.py:78]: It's not possible to get nested values from the controller's arguments. I initially tried it that way but they were returning `null` no matter what I did.

Here's some examples on other endpoints where this is the case as well 

https://github.com/wazuh/wazuh/blob/641a12b682770b840fa7218f3b38cb40cc69e1e2/api/api/controllers/agent_controller.py#L541-L543 

https://github.com/wazuh/wazuh/blob/641a12b682770b840fa7218f3b38cb40cc69e1e2/api/api/controllers/experimental_controller.py#L662-L664

File: api/api/controllers/agent_controller.py (hunk 3 above)
Comment (GGP1) [reply api/api/controllers/agent_controller.py:163]: Same as above

**Review by javiersanchz — DISMISSED** [review]
## Review

- The correct functioning was verified.
- What Gasti mentions in https://github.com/wazuh/wazuh/pull/25726/files#r1771281745 was also verified.
- From my side, LGTM!

**Review by fdalmaup — APPROVED** [review]
LGTM! Although the branch has conflicts with the master branch, please rebase it.

**Review by fdalmaup — APPROVED** [review]

**Pull request merged**